### PR TITLE
feat(cli): Add 'subtype' filter flag to 'lacework report-definitions …

### PR DESCRIPTION
**Pull request by dmurray-lacework**
…list' cmd

## Summary

Add a new filter flag to report-definitions list cmd to filter only specified clouds
`lacework report-definitions list --subtype AWS`
`lacework report-definitions list --subtype GCP`
`lacework report-definitions list --subtype Azure`

## How did you test this change?

- [x] Manually run `lacework report-definitions list --subtype <cloud>`
- [x] Unit test `TestFilterReportDefinitionsWithSubType`
- [x] Integrations Tests `TestReportDefintionsListWithSubtype` `TestReportDefintionsListJsonWithSubtype` 

## Issue

https://lacework.atlassian.net/browse/GROW-1403

### CODE REVIEW

File: cli/cmd/report_definitions.go [hunk 1]
```diff
@@ -22,12 +22,18 @@ import (
...
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 var (
+	reportDefinitionsCmdState = struct {
+		// filter report definitions by subtype. 'Aws', 'GCP' or 'Azure'
+		SubType string
+	}{}
+
```
Comment (djmctavish) [inline cli/cmd/report_definitions.go:36]: Just a single string definition for this struct, is this for future expansion?

File: integration/test_resources/help/report-definition_list [hunk 2]
```diff
@@ -7,7 +7,8 @@ Aliases:
   list, ls
 
 Flags:
-  -h, --help   help for list
+  -h, --help             help for list
+      --subtype string   filter report definitions by subtype. 'Aws', 'GCP' or 'Azure'
```
Comment (djmctavish) [inline integration/test_resources/help/report-definition_list:11]: The help for this command gives the following
`--subtype string   filter report definitions by subtype. 'Aws', 'GCP' or 'Azure'`

This does not work with Aws, Gcp they need to match the full AWS, GCP, Azure etc. Perhaps make the user input choice uppercase for the check for Gcp and Aws?

Also if the user inputs `lacework report-definition list --subtype AZURE `then the error is 
`ERROR 'AZURE' is not valid. Report definitions subtype can be [AWS Azure GCP]

**Review by djmctavish — COMMENTED** [review]
Checked out this branch and run the commands, just a few minor obbservations

**Comment by djmctavish** [comment]
I also see that the cli command line is actually `report-definition` NOT `report-definitions`, not sure that we want to change that in the CLI, possible change the command examples in the PR summary to match!

File: cli/cmd/report_definitions.go (hunk 1 above)
Comment (dmurray-lacework) [reply cli/cmd/report_definitions.go:36]: Yeh, this is the pattern we use for flags. Though you are right it is not needed right now. I'll leave it here for any potential future flags

**Comment by dmurray-lacework** [comment]
> I also see that the cli command line is actually `report-definition` NOT `report-definitions`, not sure that we want to change that in the CLI, possible change the command examples in the PR summary to match!

We have it aliased for 'report-definitions' but you're right 'report-definition' is the actual name

**Review by djmctavish — APPROVED** [review]

**Pull request merged**